File: yt/yt/flow/examples/java/shuffle/shuffle/src/main/java/tech/ytsaurus/flow/examples/shuffle/ShuffleComputationContext.java
package tech.ytsaurus.flow.examples.shuffle;

import java.util.List;

import org.springframework.context.annotation.Configuration;
import tech.ytsaurus.flow.computation.Computation;
import tech.ytsaurus.flow.computation.SourceComputation;
import tech.ytsaurus.flow.spring.ComputationProvider;

// [BEGIN computation_context]
@Configuration
public class ShuffleComputationContext implements ComputationProvider {

    @Override
    public List<Computation> getComputations() {
        return List.of(
                SourceComputation.builder()
                        .setComputationId("reader")
                        .setProcessFunction(new EventMapper())
                        .build(),
                Computation.builder()
                        .setComputationId("reducer")
                        .setProcessFunction(new EventReducer())
                        .build()
        );
    }
}
// [END computation_context]

File: yt/yt/flow/examples/kotlin/shuffle/shuffle/src/main/kotlin/tech/ytsaurus/flow/examples/shuffle/ShuffleComputationContext.kt
package tech.ytsaurus.flow.examples.shuffle

import org.springframework.context.annotation.Configuration
import tech.ytsaurus.flow.computation.Computation
import tech.ytsaurus.flow.computation.SourceComputation
import tech.ytsaurus.flow.spring.ComputationProvider

// [BEGIN computation_context]
@Configuration
open class ShuffleComputationContext : ComputationProvider {

    override fun getComputations(): List<Computation> {
        return listOf(
            SourceComputation.builder()
                .setComputationId("reader")
                .setProcessFunction(EventMapper())
                .build(),
            Computation.builder()
                .setComputationId("reducer")
                .setProcessFunction(EventReducer())
                .build(),
        )
    }
}
// [END computation_context]
